src/model/core/EthqlBlock.ts

```diff
@@ -29,11 +30,17 @@ interface TransactionsRolesArgs extends WithTransactionFilter {
 
 class EthqlBlock implements EthqlBlock {
   private static transactionFilter(filter: TransactionFilter): (tx: EthqlTransaction) => boolean {
-    return !filter || filter.withInput === undefined //
-      ? tx => true
-      : filter.withInput
+    const withInput = !filter || filter.withInput === undefined ?
+      tx => true : filter.withInput
         ? tx => !!tx.inputData
         : tx => !tx.inputData;
+
+    const contractCreation = !filter || filter.contractCreation === undefined ?
+      tx => true : filter.contractCreation
+        ? tx => tx.to.address === null
+        : tx => tx.to.address;
```

Comment:
This is mixing return types. In the last case (catch-all else), the type is (tx: EthqlTransaction) => string. Why don't you use the ! operator and the !! trick like above?

I haven't read it myself, but take a look at this: https://medium.com/@pddivine/javascript-bang-bang-i-shot-you-down-use-of-double-bangs-in-javascript-7c9d94446054

Reply:
Ah I see. SGTM

Reply:
Hmm unfortunately !! and ! don't work in this context and cause 16 tests to fail. I haven't had a chance to investigate yet so for now I will use === null and !== null as checks.